zboxcore/sdk/allocation.go

```diff
@@ -1448,3 +1458,55 @@ func (a *Allocation) getConsensuses() (fullConsensus, consensusThreshold int) {
 
 	return a.DataShards + a.ParityShards, a.DataShards + 1
 }
+
+func (a *Allocation) UpdateWithRepair(
+	size, expiry int64,
+	lock uint64,
+	updateTerms bool,
+	addBlobberId, removeBlobberId string,
+	setThirdPartyExtendable bool, fileOptionsParams *FileOptionsParameters,
+	statusCB StatusCallback,
+) (string, error) {
+
+	l.Logger.Info("Uploadating allocation")
+	hash, _, err := UpdateAllocation(size, expiry, a.ID, lock, updateTerms, addBlobberId, removeBlobberId, setThirdPartyExtendable, fileOptionsParams)
+	if err != nil {
+		return "", err
+	}
+	l.Logger.Info(fmt.Sprintf("allocation updated with hash: %s", hash))
+
+	if addBlobberId != "" {
+		l.Logger.Info("waiting for the blobber to be added to network")
+
+		for {
```

Comment:
We need to have a timeout to break the loop, or we could have infinite loop if the blobber could not be found from the allocation.

Reply:
fixed in [395679b](https://github.com/0chain/gosdk/pull/874/commits/d9b5334c2a8a78d107ac0e209fe7dbefca0faeb3
)